### ISSUE TITLE
Example from tutorial crashes

### ISSUE THREAD

**Issue by jesusha123** [jbeder/yaml-cpp#1294]
## Steps to reproduce

- Create new project in CLion with VS and the following CMake flags
  - `-DVCPKG_TARGET_TRIPLET=x64-windows-release -DCMAKE_TOOLCHAIN_FILE=C:\Users\Max\.vcpkg-clion\vcpkg\scripts\buildsystems\vcpkg.cmake`
- Create vcpkg.json file (see below)
- Create simple hello world, then add the example from here: [Building Nodes](https://github.com/jbeder/yaml-cpp/wiki/Tutorial#building-nodes)

vcpkg.json file:
```json
{
  "name" : "yaml-crash",
  "version-string" : "0.0.1",
  "builtin-baseline" : "a34c873a9717a888f58dc05268dea15592c2f0ff",
  "dependencies" : [{
    "name" : "yaml-cpp",
    "version>=" : "0.8.0#1"
  }]
}
```

main.cpp file:
```cpp
#include <iostream>
#include "yaml-cpp/yaml.h"

int main()
{
    YAML::Node node;  // starts out as null
    node["key"] = "value";  // it now is a map node
    node["seq"].push_back("first element");  // node["seq"] automatically becomes a sequence
    node["seq"].push_back("second element");

    node["mirror"] = node["seq"][0];  // this creates an alias
    node["seq"][0] = "1st element";  // this also changes node["mirror"]
    node["mirror"] = "element #1";  // and this changes node["seq"][0] - they're really the "same" node

    node["self"] = node;  // you can even create self-aliases
    node[node["mirror"]] = node["seq"];  // and strange loops :)
    std::cout << "Hello, World!" << std::endl;
    return 0;
}
```

CMakeLists.txt file:
```cmake
cmake_minimum_required(VERSION 3.28)
project(yaml_crash)

set(CMAKE_CXX_STANDARD 23)

find_package(yaml-cpp REQUIRED)

add_executable(yaml_crash main.cpp)

target_link_libraries(yaml_crash
    yaml-cpp::yaml-cpp
)
```

## Observed results

- Terminal shows the following error:

```sh
Process finished with exit code -1073741819 (0xC0000005)
```

- Debugger shows an exception when executing `node["key"] = "value";
```sh
Exception 0xc0000005 encountered at address 0x7ff6005f1d7b: Access violation reading location 0xffffffffffffffff
```

![image](https://github.com/jbeder/yaml-cpp/assets/1208947/16c62e50-771c-4789-82c6-cc8667035a09)

## Expected result

Not to crash

## System

- Windows 10
- Visual Studio 2022 Community Edition
- yaml-cpp 0.8.0#1 (from vcpkg)
- CMake

**Comment by jbeder** [comment]
This is crashing on the second line? This one:

```
node["key"] = "value";  // it now is a map node
```

If so, I strongly suspect this is an issue with your setup, since there are unit tests that cover this basic use case. Could you take a closer look, and maybe ask on Stack Overflow?

**Comment by jesusha123** [comment]
I have a CLion debug configuration, and I switched the vcpkg triplet from x64-windows-release to x64-windows.

Based on the last comment, I suspected debug configurations do not work well with the x64-windows-release vcpkg triplet. It works now, although I don't understand why. I may take a look at this in the future.

I use another library, libcurl with x64-windows-release with CLion debug configuration and it works fine.

Anyways, thank you so much for your quick response.
